libtenzir/builtins/operators/python.cpp

```diff
@@ -354,15 +378,28 @@ class plugin final : public virtual operator_plugin<python_operator> {
   }
 
   auto parse_operator(parser_interface& p) const -> operator_ptr override {
-    auto command = std::string{};
+    auto command = std::optional<std::string>{};
     auto requirements = std::string{};
+    auto filename = std::string{};
     auto parser = argument_parser{"python", "https://docs.tenzir.com/next/"
                                             "operators/transformations/python"};
     parser.add("-r,--requirements", requirements, "<requirements>");
+    parser.add("-f,--file", filename, "<filename>");
     parser.add(command, "<command>");
     parser.parse(p);
+    if (!filename.empty() == command.has_value()) {
+      diagnostic::error(
+        "Must have exactly one of the '--file' argument or inline code")
+        .throw_();
+    }
```

Comment:
If both are present, you probably want to have a different error message that refers to the location of both arguments. You can get them by making the arguments `std::optional<located<std::string>>` instead.

Generally, the error messages start lower-case, and we use backticks rather than single quotes for referencing code.